### PR TITLE
Revert "[Dialogs] Remove use of MDCFlatButton for MDCButton and MDCTextButtonThemer (#4739)"

### PR DESCRIPTION
**Pull request by codeman7**
Revert "[Dialogs] Remove use of MDCFlatButton for MDCButton and MDCTextButtonThemer (#4739)"

This reverts commit 84d5dfbb83d0118ff26921318df8cc549d919809.

This commit resulted in undesired behavior for a client, this change reverts to the old desired behavior.

### CODE REVIEW

**Review by yarneo — APPROVED** [review]

File: components/Dialogs/src/private/MDCAlertControllerView+Private.m [hunk 1]
```diff
@@ -48,9 +47,8 @@ @interface MDCAlertControllerView ()
 @end
 
 @implementation MDCAlertControllerView {
-    NSMutableArray<MDCButton *> *_actionButtons;
+    NSMutableArray<MDCFlatButton *> *_actionButtons;
     BOOL _mdc_adjustsFontForContentSizeCategory;
```
[the remark this answers is not in the document]
Comment (yarneo) [reply components/Dialogs/src/private/MDCAlertControllerView+Private.m:51]: clang don't be silly!!

**Review by jverkoey — APPROVED** [review]

File: components/Dialogs/src/private/MDCAlertControllerView+Private.m (hunk 1 above)
[the remark this answers is not in the document]
Comment (romoore) [reply components/Dialogs/src/private/MDCAlertControllerView+Private.m:51]: I thought it's only 2 space indents for ivars?

[the remark this answers is not in the document]
Comment (romoore) [reply components/Dialogs/src/private/MDCAlertControllerView+Private.m:51]: At least that's the example at the top of the style guide:

http://google.github.io/styleguide/objcguide.html

[the remark this answers is not in the document]
Comment (jverkoey) [reply components/Dialogs/src/private/MDCAlertControllerView+Private.m:51]: It is two space indents yes. This is a revert though so we're not addressing style issues.

[the remark this answers is not in the document]
Comment (codeman7) [reply components/Dialogs/src/private/MDCAlertControllerView+Private.m:51]: I think we use four throughout MDC but the style guide does seem to have two.

[the remark this answers is not in the document]
Comment (romoore) [reply components/Dialogs/src/private/MDCAlertControllerView+Private.m:51]: It's a good reason to start enforcing the style guide. :)

**Pull request merged**